### ISSUE TITLE
Report instance and class variable incompatibility for multi-inheritance

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#2936]
Pyright reports method incompatibilities when a class derives from two or more base classes, but it doesn't currently check for non-method type compatibility. The following should generate an error.

```python
# pyright: strict

class C:
    test: str = "a"

class D:
    test: int = 1

class CD(C, D):
    pass
 ```

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.229, which I just published. It will also be included in the next release of pylance.
